### DISCUSSION TITLE
Customizable variable modification

### DISCUSSION THREAD

**Discussion by Ning-S211** [vdemichev/DiaNN#1890] [Q&A]
Hi,
Recently, I wanted to search for any variable modifications of retinoylation in my data. This modification is not present in Unimod. I would like to know if the absence of this modification in the results I obtained is related to my search settings. 
Thank you!

diann.exe --f "D:\Data\20260204_LX\H10N3_NEU.raw
" --f "D:\Data\20260204_LX\H10N3_NEU1.raw
" --f "D:\Data\20260204_LX\H10N3_NEU2.raw
" --f "D:\Data\20260204_LX\H10N3_NEU3.raw
" --lib "" --threads 48 --verbose 1 --out "D:\Data\20260204_LX\NEU10\report.parquet" --qvalue 0.01 --matrices  --predictor --fasta "D:\fasta\uniprotkb_mouse_AND_reviewed_true_AND_m_2025_09_28.fasta" --fasta-search --min-fr-mz 150 --max-fr-mz 2000 --met-excision --min-pep-len 5 --max-pep-len 52 --min-pr-mz 380 --max-pr-mz 980 --min-pr-charge 1 --max-pr-charge 6 --cut K*,R* --missed-cleavages 2 --unimod4 --var-mods 2 --mass-acc 20 --mass-acc-ms1 10 --peptidoforms --reanalyse --rt-profiling --variable-modifications "Retinoylation,K,284.2;Oxidation,M,15.9949" --fixed-modifications "Carbamidomethyl,C,57.0215" --modification-max 2 --ptm-score-threshold 0.7 --enable-ptm-scoring

DIA-NN 2.2.0 Academia  (Data-Independent Acquisition by Neural Networks)
Compiled on May 29 2025 21:29:29
Current date and time: Wed Mar  4 18:45:07 2026
CPU: GenuineIntel Intel(R) Xeon(R) w9-3495X
SIMD instructions: AVX AVX2 AVX512CD AVX512F FMA SSE4.1 SSE4.2 
Logical CPU cores: 112
Thread number set to 48
Output will be filtered at 0.01 FDR
Precursor/protein x samples expression level matrices will be saved along with the main report
Deep learning will be used to generate a new in silico spectral library from peptides provided
DIA-NN will carry out FASTA digest for in silico lib generation
Min fragment m/z set to 150
Max fragment m/z set to 2000
N-terminal methionine excision enabled
Min peptide length set to 5
Max peptide length set to 52
Min precursor m/z set to 380
Max precursor m/z set to 980
Min precursor charge set to 1
Max precursor charge set to 6
In silico digest will involve cuts at K*,R*
Maximum number of missed cleavages set to 2
Cysteine carbamidomethylation enabled as a fixed modification
Maximum number of variable modifications set to 2
Peptidoform scoring enabled
MBR enabled; .quant files will only be saved to disk during the first pass
The spectral library (if generated) will retain the original spectra but will include empirically-aligned RTs
WARNING: unrecognised option [--variable-modifications Retinoylation,K,284.2;Oxidation,M,15.9949]
WARNING: unrecognised option [--fixed-modifications Carbamidomethyl,C,57.0215]
WARNING: unrecognised option [--modification-max 2]
WARNING: unrecognised option [--ptm-score-threshold 0.7]
WARNING: unrecognised option [--enable-ptm-scoring]
Mass accuracy will be fixed to 2e-05 (MS2) and 1e-05 (MS1)
WARNING: incorrect settings, the in silico-predicted library must be generated in a separate pipeline step and then used to process the raw data, now without activating FASTA digest

4 files will be processed
[0:00] Loading FASTA D:\fasta\uniprotkb_mouse_AND_reviewed_true_AND_m_2025_09_28.fasta
[0:09] Processing FASTA
[0:15] Assembling elution groups
[0:25] 4383260 precursors generated
[0:25] Gene names missing for some isoforms
[0:25] Library contains 17239 proteins, and 16904 genes
[0:31] Encoding peptides for spectra and RTs prediction
[0:47] Predicting spectra and IMs
[3:09] Predicting RTs
[3:36] Decoding predicted spectra and IMs
[3:39] Decoding RTs
[3:39] Saving the library to D:\Data\20260204_LX\NEU10\report-lib.predicted.speclib
[3:50] Initialising library
[4:11] Loading spectral library D:\Data\20260204_LX\NEU10\report-lib.predicted.speclib
[4:19] Library annotated with sequence database(s): D:\fasta\uniprotkb_mouse_AND_reviewed_true_AND_m_2025_09_28.fasta
[4:20] Spectral library loaded: 17239 protein isoforms, 55350 protein groups and 4383260 precursors in 2486919 elution groups.
[4:20] Loading protein annotations from FASTA D:\fasta\uniprotkb_mouse_AND_reviewed_true_AND_m_2025_09_28.fasta
[4:20] Annotating library proteins with information from the FASTA database
[4:20] Gene names missing for some isoforms
[4:20] Library contains 17239 proteins, and 16904 genes
[4:27] Initialising library

First pass: generating a spectral library from DIA data

[4:47] File #1/4
[4:47] Loading run D:\Data\20260204_LX\H10N3_NEU.raw
[5:04] Pre-processing...
[5:06] 1348 MS1 and 135713 MS2 scans in 450 (inferred) and 225 (encoded) cycles, 4383255 precursors in range
[5:08] Calibrating with mass accuracies 22 (MS1), 25 (MS2)
[6:01] RT window set to 1.12225
[6:01] Peak width: 3.36
[6:01] Scan window radius set to 7
[6:01] Recommended MS1 mass accuracy setting: 4.5 ppm
[6:40] Searching decoys
[7:13] Main search
[8:22] Removing low confidence identifications
[8:30] Removing interfering precursors
[8:35] Training neural networks on 32758 target and 17145 decoy PSMs
[8:50] Training neural networks on 32758 target and 16467 decoy PSMs
[9:01] Number of IDs at 0.01 FDR: 16737
[9:02] Precursors at 1% peptidoform FDR: 15811
[9:02] Calculating protein q-values
[9:03] Number of genes identified at 1% FDR: 3650 (precursor-level), 3229 (protein-level) (inference performed using proteotypic peptides only)
[9:03] Quantification
[9:05] Quantification information saved to D:\Data\20260204_LX\H10N3_NEU.raw.quant

[9:05] File #2/4
[9:05] Loading run D:\Data\20260204_LX\H10N3_NEU1.raw
[9:23] Pre-processing...
[9:25] 1347 MS1 and 135932 MS2 scans in 449 (inferred) and 225 (encoded) cycles, 4383255 precursors in range
[9:26] Calibrating with mass accuracies 23 (MS1), 25 (MS2)
[10:19] RT window set to 0.623148
[10:19] Recommended MS1 mass accuracy setting: 4.2 ppm
[10:52] Searching decoys
[11:23] Main search
[12:28] Removing low confidence identifications
[12:39] Removing interfering precursors
[12:44] Training neural networks on 54443 target and 31398 decoy PSMs
[13:05] Training neural networks on 54443 target and 30823 decoy PSMs
[13:23] Number of IDs at 0.01 FDR: 22558
[13:24] Precursors at 1% peptidoform FDR: 21403
[13:24] Calculating protein q-values
[13:25] Number of genes identified at 1% FDR: 3610 (precursor-level), 3149 (protein-level) (inference performed using proteotypic peptides only)
[13:25] Quantification
[13:28] Quantification information saved to D:\Data\20260204_LX\H10N3_NEU1.raw.quant

[13:28] File #3/4
[13:28] Loading run D:\Data\20260204_LX\H10N3_NEU2.raw
[13:46] Pre-processing...
[13:48] 1345 MS1 and 135854 MS2 scans in 449 (inferred) and 225 (encoded) cycles, 4383255 precursors in range
[13:49] Calibrating with mass accuracies 23 (MS1), 25 (MS2)
[14:41] RT window set to 0.709546
[14:42] Recommended MS1 mass accuracy setting: 4.5 ppm
[15:17] Searching decoys
[15:52] Main search
[17:05] Removing low confidence identifications
[17:16] Removing interfering precursors
[17:21] Training neural networks on 53812 target and 31937 decoy PSMs
[17:42] Training neural networks on 53812 target and 30824 decoy PSMs
[18:00] Number of IDs at 0.01 FDR: 22938
[18:01] Precursors at 1% peptidoform FDR: 21744
[18:01] Calculating protein q-values
[18:02] Number of genes identified at 1% FDR: 3567 (precursor-level), 3140 (protein-level) (inference performed using proteotypic peptides only)
[18:02] Quantification
[18:05] Quantification information saved to D:\Data\20260204_LX\H10N3_NEU2.raw.quant

[18:05] File #4/4
[18:05] Loading run D:\Data\20260204_LX\H10N3_NEU3.raw
[18:23] Pre-processing...
[18:25] 1346 MS1 and 135938 MS2 scans in 449 (inferred) and 225 (encoded) cycles, 4383255 precursors in range
[18:27] Calibrating with mass accuracies 22 (MS1), 25 (MS2)
[19:21] RT window set to 0.66716
[19:22] Recommended MS1 mass accuracy setting: 4 ppm
[19:58] Searching decoys
[20:33] Main search
[21:43] Removing low confidence identifications
[21:55] Removing interfering precursors
[22:00] Training neural networks on 58231 target and 35153 decoy PSMs
[22:25] Training neural networks on 58231 target and 33952 decoy PSMs
[22:45] Number of IDs at 0.01 FDR: 23702
[22:45] Precursors at 1% peptidoform FDR: 22363
[22:46] Calculating protein q-values
[22:46] Number of genes identified at 1% FDR: 3712 (precursor-level), 3254 (protein-level) (inference performed using proteotypic peptides only)
[22:47] Quantification
[22:50] Quantification information saved to D:\Data\20260204_LX\H10N3_NEU3.raw.quant

[22:50] Cross-run analysis
[22:50] Reading quantification information: 4 files
[23:17] Quantifying peptides
[24:14] Assembling protein groups
[24:18] Quantifying proteins
[24:18] Calculating q-values for protein and gene groups
[24:21] Calculating global q-values for protein and gene groups
[24:21] Protein groups with global q-value <= 0.01: 3907
[24:22] Compressed report saved to D:\Data\20260204_LX\NEU10\report-first-pass.parquet. Use R 'arrow' or Python 'PyArrow' package to process
[24:22] Saving precursor levels matrix
[24:22] Precursor levels matrix (1% precursor and protein group FDR) saved to D:\Data\20260204_LX\NEU10\report-first-pass.pr_matrix.tsv.
[24:22] Manifest saved to D:\Data\20260204_LX\NEU10\report-first-pass.manifest.txt
[24:22] Stats report saved to D:\Data\20260204_LX\NEU10\report-first-pass.stats.tsv
[24:22] Generating spectral library:
[24:24] 30868 target and 310 decoy precursors saved
[24:24] Spectral library saved to D:\Data\20260204_LX\NEU10\report-lib.parquet

[24:25] Loading spectral library D:\Data\20260204_LX\NEU10\report-lib.parquet
[24:26] Spectral library loaded: 4742 protein isoforms, 4676 protein groups and 31178 precursors in 27031 elution groups.
[24:26] Loading protein annotations from FASTA D:\fasta\uniprotkb_mouse_AND_reviewed_true_AND_m_2025_09_28.fasta
[24:27] Annotating library proteins with information from the FASTA database
[24:27] Gene names missing for some isoforms
[24:27] Library contains 4742 proteins, and 4707 genes
[24:27] Initialising library
[24:30] Saving the library to D:\Data\20260204_LX\NEU10\report-lib.parquet.skyline.speclib

Second pass: using the newly created spectral library to reanalyse the data

[24:30] File #1/4
[24:30] Loading run D:\Data\20260204_LX\H10N3_NEU.raw
[24:45] Pre-processing...
[24:46] 1348 MS1 and 135713 MS2 scans in 450 (inferred) and 225 (encoded) cycles, 30868 precursors in range
[24:46] Calibrating with mass accuracies 22 (MS1), 25 (MS2)
[24:46] RT window set to 1.06714
[24:47] Recommended MS1 mass accuracy setting: 7 ppm
[24:47] Searching decoys
[24:48] Main search
[24:51] Removing low confidence identifications
[24:54] Removing interfering precursors
[24:54] Training neural networks on 23105 target and 13025 decoy PSMs
[25:03] Training neural networks on 23095 target and 13403 decoy PSMs
[25:12] Number of IDs at 0.01 FDR: 18756
[25:12] Precursors at 1% peptidoform FDR: 18243
[25:12] Calculating protein q-values
[25:12] Number of genes identified at 1% FDR: 3676 (precursor-level), 3153 (protein-level) (inference performed using proteotypic peptides only)
[25:12] Quantification

[25:14] File #2/4
[25:14] Loading run D:\Data\20260204_LX\H10N3_NEU1.raw
[25:35] Pre-processing...
[25:36] 1347 MS1 and 135932 MS2 scans in 449 (inferred) and 225 (encoded) cycles, 30868 precursors in range
[25:36] Calibrating with mass accuracies 23 (MS1), 25 (MS2)
[25:36] RT window set to 0.162154
[25:36] Recommended MS1 mass accuracy setting: 7 ppm
[25:37] Searching decoys
[25:38] Main search
[25:40] Removing low confidence identifications
[25:43] Removing interfering precursors
[25:44] Training neural networks on 25855 target and 13642 decoy PSMs
[25:53] Training neural networks on 25848 target and 15128 decoy PSMs
[26:03] Number of IDs at 0.01 FDR: 24248
[26:03] Precursors at 1% peptidoform FDR: 23829
[26:03] Calculating protein q-values
[26:04] Number of genes identified at 1% FDR: 3674 (precursor-level), 3141 (protein-level) (inference performed using proteotypic peptides only)
[26:04] Quantification

[26:07] File #3/4
[26:07] Loading run D:\Data\20260204_LX\H10N3_NEU2.raw
[26:28] Pre-processing...
[26:29] 1345 MS1 and 135854 MS2 scans in 449 (inferred) and 225 (encoded) cycles, 30868 precursors in range
[26:29] Calibrating with mass accuracies 23 (MS1), 25 (MS2)
[26:29] RT window set to 0.150804
[26:30] Recommended MS1 mass accuracy setting: 7 ppm
[26:30] Searching decoys
[26:31] Main search
[26:33] Removing low confidence identifications
[26:37] Removing interfering precursors
[26:38] Training neural networks on 25792 target and 13708 decoy PSMs
[26:47] Training neural networks on 25787 target and 15182 decoy PSMs
[26:57] Number of IDs at 0.01 FDR: 24202
[26:57] Precursors at 1% peptidoform FDR: 23786
[26:57] Calculating protein q-values
[26:57] Number of genes identified at 1% FDR: 3633 (precursor-level), 3169 (protein-level) (inference performed using proteotypic peptides only)
[26:57] Quantification

[27:00] File #4/4
[27:00] Loading run D:\Data\20260204_LX\H10N3_NEU3.raw
[27:21] Pre-processing...
[27:22] 1346 MS1 and 135938 MS2 scans in 449 (inferred) and 225 (encoded) cycles, 30868 precursors in range
[27:22] Calibrating with mass accuracies 23 (MS1), 25 (MS2)
[27:23] RT window set to 0.150969
[27:23] Recommended MS1 mass accuracy setting: 6 ppm
[27:23] Searching decoys
[27:24] Main search
[27:26] Removing low confidence identifications
[27:29] Removing interfering precursors
[27:30] Training neural networks on 25704 target and 13466 decoy PSMs
[27:39] Training neural networks on 25699 target and 15088 decoy PSMs
[27:49] Number of IDs at 0.01 FDR: 24326
[27:49] Precursors at 1% peptidoform FDR: 23968
[27:49] Calculating protein q-values
[27:49] Number of genes identified at 1% FDR: 3701 (precursor-level), 3152 (protein-level) (inference performed using proteotypic peptides only)
[27:49] Quantification

[27:52] Cross-run analysis
[27:52] Reading quantification information: 4 files
[27:53] Quantifying peptides
WARNING: QuantUMS requires 6 or more runs for the optimisation of its hyperparameters to perform best.
[28:45] Quantification parameters: 0.390641, 0.00440435, 0.00147485, 0.0118589, 0.0122768, 0.0123197, 0.0936952, 0.151448, 0.115637, 0.0182684, 0.0486694, 0.0426818, 0.338617, 0.0500717, 0.0538547, 0.0108616
[29:07] Quantifying proteins
[29:07] Calculating q-values for protein and gene groups
[29:07] Calculating global q-values for protein and gene groups
[29:07] Protein groups with global q-value <= 0.01: 3730
[29:08] Compressed report saved to D:\Data\20260204_LX\NEU10\report.parquet. Use R 'arrow' or Python 'PyArrow' package to process
[29:08] Saving precursor levels matrix
[29:08] Precursor levels matrix (1% precursor and protein group FDR) saved to D:\Data\20260204_LX\NEU10\report.pr_matrix.tsv.
[29:08] Saving protein group levels matrix
[29:08] Protein groups matrix saved to D:\Data\20260204_LX\NEU10\report.pg_matrix.tsv.
[29:08] Saving gene group levels matrix
[29:08] Gene groups matrix saved to D:\Data\20260204_LX\NEU10\report.gg_matrix.tsv.
[29:08] Saving unique genes levels matrix
[29:08] Unique genes matrix saved to D:\Data\20260204_LX\NEU10\report.unique_genes_matrix.tsv.
[29:08] Manifest saved to D:\Data\20260204_LX\NEU10\report.manifest.txt
[29:08] Stats report saved to D:\Data\20260204_LX\NEU10\report.stats.tsv

The following warnings or errors (in alphabetic order) were detected at least the indicated number of times:
WARNING: QuantUMS requires 6 or more runs for the optimisation of its hyperparameters to perform best. : 1
WARNING: incorrect settings, the in silico-predicted library must be generated in a separate pipeline step and then used to process the raw data, now without activating FASTA digest : 1
WARNING: unrecognised option [--enable-ptm-scoring] : 1
WARNING: unrecognised option [--fixed-modifications Carbamidomethyl,C,57.0215] : 1
WARNING: unrecognised option [--modification-max 2] : 1
WARNING: unrecognised option [--ptm-score-threshold 0.7] : 1
WARNING: unrecognised option [--variable-modifications Retinoylation,K,284.2;Oxidation,M,15.9949] : 1
Finished

**Comment by vdemichev** [comment]
Hi,

There's no need for the modification to be present in UniMod. However, in the log you can see many warnings from DIA-NN, in particular syntax is incorrect and the workflow is not configured correctly.

Best,
Vadim
